### PR TITLE
feat:(react-nav-preview): Expanding Drawer types

### PR DESCRIPTION
**Pull request by mltejera**
Currently we're only able to use the Inline flavor of drawer. Design wants both options - Overly in the case of a narrow screen. The main goal of this component is to consolidate the following code from:

```
<Nav ...some Nav props>
   <StylingProvider> // still up for discussion if this is necessary/recommended
       <Drawer ...some  drawer props>
           <DrawerHeaderAndChildren/>
           <DrawerBodyAndChildren/>
           <DrawerFooterAndChildren/>
``` 

To

```
<NavDrawer  ...some Nav, drawer and styling props>
   <DrawerHeaderAndChildren/>
   <DrawerBodyAndChildren/>
   <DrawerFooterAndChildren/>
```

Had to disable a boatload of tests because the underlying Drawer/Modal component caused a lot of shenanigans. 

Ladders to #26649

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-nav-preview/src/components/NavDrawer/NavDrawer.types.ts
Comment (mltejera) [inline packages/react-components/react-nav-preview/src/components/NavDrawer/NavDrawer.types.ts:13]: this feels a little yucky and loose. I'd prefer it if drawer exported the type for me to import.

File: packages/react-components/react-nav-preview/stories/NavDrawerInline/index.stories.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/stories/NavDrawerInline/index.stories.tsx:8]: Imo, both the stories from Overlay vs Inline are variations of NavDrawer, so they should live in a NavDrawer story like this:
- stories
  - NavDrawer
    - index.stories.tsx
    - NavDrawerOverlay.stories.tsx
    - NavDrawerInline.stories.tsx

This is because when it's moved out of preview, you'll be taking up 4 component spaces in the docs. Ideally all Nav related stories should live under one place. Alternatively, you could do something like Button and have
- Nav
  - Nav
  - NavDrawer

File: packages/react-components/react-nav-preview/stories/NavDrawerOverlay/NavDrawerOverlay.stories.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/stories/NavDrawerOverlay/NavDrawerOverlay.stories.tsx:27]: We talked offline about this, but I've pulled this PR and adding a button to trigger the drawer does work. This should remove the need for useIsSSR.

One thing to note as well is that I don't think we should use `useIsSSR` inside stories. This is test specific logic that should only be done within the test. Therefore Before this PR goes in we should remove this. Since we are covering this behind a button (if no other issues come up), we don't have to disable the story from `test-ssr` anymore.

File: scripts/test-ssr/src/commands/main.ts
Comment (sopranopillow) [inline scripts/test-ssr/src/commands/main.ts:26]:
```suggestion
```

File: packages/react-components/react-nav-preview/src/components/NavDrawer/NavDrawer.test.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/src/components/NavDrawer/NavDrawer.test.tsx:6]:
```suggestion
/**
   * Why these tests are disabled:
   * component-handles-ref|component-has-root-ref: Drawer (NavDrawer's base) uses the Dialog under the hood and Dialog do not handle ref, as it is a renderless component
   * component-handles-classname|component-has-static-classnames-object|make-styles-overrides-win: Drawer uses the DialogSurface component to render the classname, so the main component do not handle classname.
   * consistent-callback-args: Disabled that as the Drawer callback function uses the same signature as the Dialog, and Dialog has those tests disabled.
   */
```

File: packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx:18]: We could remove the need for assert slots if we do this instead:
```ts
{renderDrawer(state as DrawerState)}
```

you should test this locally as well though

File: packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx:14]: If you go with my other comment
```suggestion
```

Comment (sopranopillow) [reply packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx:14]: And you would also remove the jsxRuntime comments on the top

File: packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx:5]: This one should go at the bottom

File: packages/react-components/react-nav-preview/src/components/NavItem/useNavItem.ts
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/src/components/NavItem/useNavItem.ts:46]: This should not be needed for any of the components, I think the arrow navigation might be messing it up. We should debug this rather than add tabIndex 0 since <a> is a tabstop by default. Feel free to ping me offline to check it out

**Review by sopranopillow — COMMENTED** [review]
Not blocking the PR, but the button and test changes should be added before merging

File: packages/react-components/react-nav-preview/src/components/NavItem/useNavItem.ts
Comment (mltejera) [reply packages/react-components/react-nav-preview/src/components/NavItem/useNavItem.ts:46]: Yep yep, makes sense.

Created this issue to investigate further in a seperate PR.

https://github.com/microsoft/fluentui/issues/31190

File: packages/react-components/react-nav-preview/stories/NavDrawerOverlay/NavDrawerOverlay.stories.tsx
Comment (mltejera) [reply packages/react-components/react-nav-preview/stories/NavDrawerOverlay/NavDrawerOverlay.stories.tsx:27]: ten four - removed!

File: packages/react-components/react-nav-preview/src/components/NavDrawer/NavDrawer.test.tsx
Comment (mltejera) [reply packages/react-components/react-nav-preview/src/components/NavDrawer/NavDrawer.test.tsx:6]: replaced - thanks!

File: packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx
Comment (mltejera) [reply packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx]: changed, tested and pushed - good call!

File: packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx
Comment (mltejera) [reply packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx]: 🫡

File: packages/react-components/react-nav-preview/stories/NavDrawerInline/index.stories.tsx
Comment (mltejera) [reply packages/react-components/react-nav-preview/stories/NavDrawerInline/index.stories.tsx]: Agreed! I'll rejigger the story structure next week when I start adding it to the main story book.

File: packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx
Comment (mltejera) [reply packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx]: I ended up having to add the comments back to pass the CI builds.

**Review by marcosmoura — APPROVED** [review]
Approved for v-build team.

File: packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx:2]: These can be removed
```suggestion
```

File: packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx:5]: Should this one be a type import?
```suggestion
import type { NavContextValues } from '../NavContext.types';
```

File: packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx:4]: I think the state also needs to be splitted into type import?
```suggestion
import { renderDrawer_unstable } from '@fluentui/react-drawer';
import type { DrawerState } from '@fluentui/react-drawer';
```

File: packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawer.stories.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawer.stories.tsx:27]: Should this be a token? I think it's spacing Large:
```suggestion
    ...shorthands.padding(tokens.spacingVerticalL, tokens.spacingHorizontalL),
```

File: packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawer.stories.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawer.stories.tsx:17]: Should the color be a token? the stroke width should also be a token, I think it's strokeWidthThin(?)

File: packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawer.stories.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawer.stories.tsx:46]: Any reason to rename the file? We usually try to keep the name as `<component><story goal>` in the case for default `<Component>Default`

File: packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawer.stories.tsx
Comment (mltejera) [reply packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawer.stories.tsx:27]: eh. Brandon and I are operating under the assumption that spacing tokens will go away when we roll out the new token strategy in a few months. They never work quite as well as people hope and end up causing a LOT of off by 1 errors.

File: scripts/test-ssr/src/commands/main.ts
Comment (mltejera) [reply scripts/test-ssr/src/commands/main.ts:25]: good catch - no idea how that got in there.

File: packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx
Comment (mltejera) [reply packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx:2]: I get this error when I do that:

image

File: packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawer.stories.tsx
Comment (mltejera) [reply packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawer.stories.tsx:46]: I think I was just tidying some file conflicts. Will rename back.

File: packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx
Comment (mltejera) [reply packages/react-components/react-nav-preview/src/components/NavDrawer/renderNavDrawer.tsx]: recapping offline - skipping this for now.

File: packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawerDefault.stories.tsx
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/stories/NavDrawer/NavDrawerDefault.stories.tsx:12]:
```suggestion
import { Button, DrawerBody, Label, Radio, RadioGroup, makeStyles, shorthands, tokens, useId } from '@fluentui/react-components';
import type { DrawerProps } from '@fluentui/react-components';
```

**Review by sopranopillow — APPROVED** [review]

**Pull request merged**
